### ISSUE TITLE
Cannot communicate with deis pods

### ISSUE THREAD

**Issue by deinspanjer** [TheNewNormal/kube-cluster-osx#48]
After getting what I think is a clean install of kube-cluster, I ran the install_deis command.  At the very end, it failed with trying to register the admin user.  It appears the hostnames it is trying to use aren't resolving properly.  I believe this is because it is trying to resolve via my LAN dns server instead of the k8s one:

```
Registering Deis Workflow user ...
http://deis.192.168.64.3.nip.io/v2/ does not appear to be a valid Deis controller.
Make sure that the Controller URI is correct, the server is running and
your client version is correct.
Error: Get http://deis.192.168.64.3.nip.io/v2/: dial tcp: lookup deis.192.168.64.3.nip.io on 192.168.1.1:53: no such host
```

I wasn't sure about the whole nip.io thing.  I read the docs on the DNS addon that you link to in your readme ( https://github.com/kubernetes/kubernetes/blob/release-1.2/cluster/addons/dns/README.md ), and it talked about using the 10.100.0.10 DNS server.  I followed the instructions on that page to run a simple busybox pod and tested out the names found there, but even with those, I still can't get deis to register.  I'm assuming that is because the 10.100.0.10 address is only visible inside the VMs:

```
bash-3.2$ deis register
Usage: deis auth:register <controller> [options]

bash-3.2$ kubectl exec busybox -- nslookup deis-router.deis
Server:    10.100.0.10
Address 1: 10.100.0.10

Name:      deis-router.deis
Address 1: 10.100.138.0
bash-3.2$ kubectl exec busybox -- nslookup deis-controller.deis
Server:    10.100.0.10
Address 1: 10.100.0.10

Name:      deis-controller.deis
Address 1: 10.100.69.107

bash-3.2$ host deis-controller.deis
Host deis-controller.deis not found: 3(NXDOMAIN)

bash-3.2$ deis auth:register deis-controller.deis
http://deis-controller.deis/v2/ does not appear to be a valid Deis controller.
Make sure that the Controller URI is correct, the server is running and
your client version is correct.

Error: Get http://deis-controller.deis/v2/: dial tcp: lookup deis-controller.deis on **192.168.1.1:53**: no such host
bash-3.2$
```

Is this a botched configuration on my end or is there some sort of conflict with my LAN DNS?

**Comment by rimusz** [comment]
@deinspanjer the DNS addon is purely for kubernetes use, e.g. it allows to reach other pods via service name.
nip.io is used as there is no local domain to associate with deis.. platform.
It has to be your LAN DNS end.

can you check that all deis workflow components are running properly with:
kubectl --namespace=deis get pods

**Comment by deinspanjer** [comment]
Well, all of the deis pods were running when I was working with this yesterday, but today, (after a sleep and restore of my MBP) when I run get pods, I see there are a few that are not running properly:

```
bash-3.2$ kubectl logs deis-workflow-manager-smsw1
Error from server: pods "deis-workflow-manager-smsw1" not found
bash-3.2$ kubectl --namespace=deis get pods
NAME                          READY     STATUS             RESTARTS   AGE
deis-builder-4v82h            0/1       Running            3          13h
deis-controller-mvj38         1/1       Running            4          13h
deis-database-7o6gd           1/1       Running            2          13h
deis-logger-fluentd-gtvrw     0/1       CrashLoopBackOff   4          13h
deis-logger-fluentd-h7hg4     0/1       CrashLoopBackOff   4          13h
deis-logger-z4xc7             1/1       Running            0          13h
deis-minio-hud5q              1/1       Running            0          13h
deis-monitor-grafana-otexz    1/1       Running            0          13h
deis-monitor-influxdb-r41dy   1/1       Running            0          13h
deis-monitor-stdout-f4gs0     1/1       Running            0          13h
deis-monitor-telegraf-rvrl0   1/1       Running            1          13h
deis-monitor-telegraf-ynp0q   1/1       Running            1          13h
deis-registry-nafns           1/1       Running            0          13h
deis-router-x7wix             0/1       CrashLoopBackOff   6          13h
deis-workflow-manager-smsw1   1/1       Running            0          13h
bash-3.2$ kubectl describe deis-router-x7wix
the server doesn't have a resource type "deis-router-x7wix"
bash-3.2$ kubectl describe pods deis-router-x7wix
Error from server: pods "deis-router-x7wix" not found
```

And further, the error I reported in issue 47 about the x509 error is back as well. :/  It seems that my problem was not in fact with the symlinked id_rsa key.

**Comment by rimusz** [comment]
@deinspanjer the problems above are not really related to the the Kube (solo and cluster) apps, I will report the issues back to my team

**Comment by rimusz** [comment]
@deinspanjer the problem is fixed with the latest Workflow v2.1.
Use v0.3.4 App which supports Workflow v2.1 install

**Comment by rimusz** [comment]
still an issue on k8s v1.3 and workflow v2.1, the issue should be fixed in workflow v2.2 by https://github.com/deis/builder/pull/382
